This is an adapted version of the original by Zoong Pham.

To use this patch, run these commands for a successful build:

    patch -p1 <patches/tru64.diff
    ./configure                          (optional if already run)
    make

based-on: 7fb08531e0ee267e4a41e209be4cb5a24d461a2d
diff --git a/syscall.c b/syscall.c
--- a/syscall.c
+++ b/syscall.c
@@ -23,6 +23,7 @@
 #include "rsync.h"
 
 #if !defined MKNOD_CREATES_SOCKETS && defined HAVE_SYS_UN_H
+#define _SOCKADDR_LEN
 #include <sys/un.h>
 #endif
 #ifdef HAVE_SYS_ATTR_H
